java/src/main/proto/skein.proto

```diff
@@ -140,6 +140,13 @@ message Master {
 }
 
 
+message CredentialProviderSpec {
+  string name = 1;
+  string uri = 2;
+  string principal = 3;
```

Comment:
Is this sufficient for any other credential provider we may want to support later?

Reply:
as explained in the big comment, having these 2 are not enough for all providers, and I switched to a map of configurations